src/main/java/javaslang/collection/Iterator.java

```diff
@@ -166,6 +167,16 @@ public T next() {
         };
     }
 
+    @Override
+    default Iterator<T> iterator() {
+        return this;
```

Comment:
I meant, if we have an Iterable, we just call myIterable.iterator() to obtain an iterator...